File: src/219. Contains Duplicate II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */

export function containsNearbyDuplicate(nums, k) {
  const map = {}
  for (let i = 0; i < nums.length; i++) {
    if (map[nums[i]] >= 0 && i - map[nums[i]] <= k) return true
    map[nums[i]] = i
  }
  return false
}

File: src/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        s = set()
        k = min(k, n - 1)
        for i in range(n):
            if i > k: s.remove(nums[i - k - 1])
            if nums[i] in s: return True
            s.add(nums[i])
        return False
            
            
        